### ISSUE TITLE
No compilation Error when defining method's return value

### ISSUE THREAD

**Issue by AdirNoyman** [Microsoft/TypeScript#28779]
I ran the following code (in Visual Studio code editor) and didn't get a compilation error (for the "deposit" method), even though the method's return value definition in the object type definition doesn't correlate to the return type that was defined in the method itself.

**TypeScript Version:**  3.2.0-dev.201xxxxx

**Search Terms:**

**Code**

type BankAccount = { money: number, deposit: (val: number) => void };

let bankAccount: BankAccount = {
    money: 2000,
    deposit(value: number): number {
        return this.money += value;
        
    }
};

let myself: { name: string, bankAccount: BankAccount, hobbies: string[] } = {
    name: "Adir",
    bankAccount: bankAccount,
    hobbies: ["Sports", "Cooking"]
};

console.log(myself.bankAccount.deposit(3000));
```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
Compilation error for the "deposit" method (returns a number when in the objects ("bankAccount") definition it wasn't supposed to return anything (void).
**Actual behavior:**
No compilation error was raised and I got output in the browser.

**Playground Link:** 

**Related Issues:**

**Comment by weswigham** [comment]
All return types are assignable to a function with a `void` return type, as `void` indicates the return type is unused. There are lint rules to make this stricter in your own codebase.
